x/pairing/keeper/staking.go

```diff
@@ -98,11 +80,22 @@ func (k Keeper) StakeNewEntry(ctx sdk.Context, creator, chainID string, amount s
 			{Key: "stake", Value: amount},
 		}
 		details = append(details, utils.Attribute{Key: "moniker", Value: moniker})
+
 		if amount.IsGTE(existingEntry.Stake) {
+			// we dont change stakeAppliedBlocks and chain once they are set, if they need to change, unstake first
+			existingEntry.Geolocation = geolocation
+			existingEntry.Endpoints = endpointsVerified
+			existingEntry.Moniker = moniker
+			existingEntry.DelegateCommission = delegationCommission
+			existingEntry.DelegateLimit = delegationLimit
+
+			k.epochStorageKeeper.ModifyStakeEntryCurrent(ctx, chainID, existingEntry, indexInStakeStorage)
+
 			// support modifying with the same stake or greater only
 			if !amount.Equal(existingEntry.Stake) {
-				// needs to charge additional tokens
-				err := verifySufficientAmountAndSendToModule(ctx, k, senderAddr, amount.Sub(existingEntry.Stake))
+				// delegate the difference
+				amount = amount.Sub(existingEntry.Stake)
```

Comment:
better practice to create a new local amount for the diff

Reply:
done